packages/sdk/migrations/src/migrations.ts

```diff
@@ -47,21 +52,20 @@ export class Migrations {
       return false;
     }
 
+    this._state.running.push(space.key.toHex());
     if (targetIndex > currentIndex) {
       const migrations = this.migrations.slice(currentIndex, targetIndex);
       for (const migration of migrations) {
-        await migration.up({ space });
-        space.properties[this.versionProperty] = migration.version;
-      }
-    } else {
-      const migrations = this.migrations.slice(targetIndex, currentIndex);
-      migrations.reverse();
-      for (const migration of migrations) {
-        await migration.down({ space });
-        const index = this.migrations.indexOf(migration);
-        space.properties[this.versionProperty] = this.migrations[index - 1]?.version;
+        const builder = new MigrationBuilder(space);
+        await migration.next({ space, builder });
+        await builder.changeProperties((propertiesStructure) => {
+          invariant(this.versionProperty, 'Migrations namespace not set');
+          propertiesStructure.data[this.versionProperty] = migration.version;
+        });
+        await builder._commit();
```

Comment:
rename to `commit`

Reply:
any reason why? generally we've prefixed internal/private apis with `_`